pkg/pool-linear/test/AaveLinearPool.test.ts

```diff
@@ -82,15 +85,27 @@ describe('AaveLinearPool', function () {
   });
 
   describe('getWrappedTokenRate', () => {
-    it('returns the expected value', async () => {
-      // Reserve's normalised income is stored with 27 decimals (i.e. a 'ray' value)
-      // 1e27 implies a 1:1 exchange rate between main and wrapped token
-      await mockLendingPool.setReserveNormalizedIncome(bn(1e27));
-      expect(await pool.getWrappedTokenRate()).to.be.eq(fp(1));
-
-      // We now double the reserve's normalised income to change the exchange rate to 2:1
-      await mockLendingPool.setReserveNormalizedIncome(bn(2e27));
-      expect(await pool.getWrappedTokenRate()).to.be.eq(fp(2));
+    context('under normal operation', () => {
+      it('returns the expected value', async () => {
+        // Reserve's normalised income is stored with 27 decimals (i.e. a 'ray' value)
+        // 1e27 implies a 1:1 exchange rate between main and wrapped token
+        await mockLendingPool.setReserveNormalizedIncome(bn(1e27));
+        expect(await pool.getWrappedTokenRate()).to.be.eq(fp(1));
+
+        // We now double the reserve's normalised income to change the exchange rate to 2:1
+        await mockLendingPool.setReserveNormalizedIncome(bn(2e27));
+        expect(await pool.getWrappedTokenRate()).to.be.eq(fp(2));
+      });
+    });
+
+    context('when Aave reverts maliciously', () => {
+      sharedBeforeEach('make Aave lending pool start reverting', async () => {
+        await mockLendingPool.setRevertMaliciously(true);
+      });
```

Comment:
It'd be good to also have a rebalancer test that fails if the lending pool is malicious. 

Also, it'd be great if we tested both swap and joinexit queries. I know we test both at the lib, but it shouldn't be too much work to have both, and it'll cover all cases.